spec/ics-007-tendermint-client/README.md

```diff
@@ -128,10 +128,10 @@ Tendermint client validity checking uses the bisection algorithm described in th
 function checkValidityAndUpdateState(
   clientState: ClientState,
   header: Header) {
-    // assert trusting period has not yet passed
+    // assert trusting period has not yet passed. This should fatally terminate a connection.
     assert(currentTimestamp() - clientState.latestTimestamp < clientState.trustingPeriod)
-    // assert header timestamp is not in the future (& transitively that is not past the trusting period)
-    assert(header.timestamp <= currentTimestamp())
+    // assert header timestamp is less than trust period in the future. This should be resolved with an intermediate header.
```

Comment:
Right, although it's worth noting that the only case when this check would fail when the first succeeds is when the chain which we're running a light client of is running more quickly than the local chain, so it should be an unusual case (this is not the same as bisection, where we need intermediate headers for a different reason).